build_tools/cmake/ctest_all.sh

```diff
@@ -61,13 +65,32 @@ fi
 IFS=',' read -ra extra_label_exclude_args <<< "${IREE_EXTRA_COMMA_SEPARATED_CTEST_LABELS_TO_EXCLUDE:-}"
 label_exclude_args+=(${extra_label_exclude_args[@]})
 
-# Join on "|"
-label_exclude_regex="($(IFS="|" ; echo "${label_exclude_args[*]}"))"
+important_ctest_args=()
```

Comment:
This isn't a *super* great name. We can just put all the args in an array. I don't think much is lost by putting everything in the array. You can even `set -x` before the invocation so it just prints out the whole command rather than doing custom printing

Reply:
I meant "important" not so much in its real-english sense as in a fairly widespread sense of "singled out as worth logging".

`set -x` does sound like a good way to achieve that, but note that we would still need to put together that array of args in order to generate the command line, so the only thing we'd be saving would be the single line doing the `echo`.  I thought since we had to generate this array anyway, we might as well use it to generate neater logging, but no strong opinion.

Reply:
It's also possible to unconditionally add all these flags, with the right default values such as `.*` for the regex.  I tried to avoid making command lines more complicated for everyone, just for my minority use case. Also, knowing CTest, it's hard to trust that there are such neat invariants as "adding this flag makes no difference at all". For example, since `--repeat-until-fail` has nontrivial side effects in terms of parallelization in general, I don't trust by default that it has no side effect when the repeat count is 1.

Reply:
I think logging the whole command line is usually worth it :-) That's copy-pasteable. I agree on not putting in flags if they're not set, especially given ctest's horrible flag handling in general

Reply:
Got it, thanks for caring :-) PR coming.

Reply:
#10136